docs/docs/vscode-setup.md

```diff
@@ -0,0 +1,71 @@
+---
+title: "Setting up Visual Studio Code for Gatsby development"
+---
+
+VS Code is a wonderful tool that can require some initial setup to get everything working. Here's some recommendations to get code flowing smoothly.
+
+## Initial Setup
+
+### .gitignore
+
+You should add the `.vscode` folder to your `.gitignore`, as your workspace settings and perferences should not be committed to a git repository. Consider setting up an `.editorconfig` file if you need to sync different settings like linting preferences across editors.
+
+### Correcting IntelliSense
+
+You should add the `typescript` package as a global or dev dependency for your project so that you can benefit from the TypeScript-driven refactoring and other development goodies.
+
+`yarn global add typescript` will add it globally.
```

Comment:
I use Code + IntelliSense without needing to install this package globally or locally - as far as I'm aware, as long as you install the JS extensions in Code then the rest is handled automatically; is that not the case?

Reply:
> lots of the various features like autoclosing of JSX tags requires TypeScript 3.0+ in the workspace

I think this feature was added in the [July update](https://code.visualstudio.com/updates/v1_26#_jsx-tag-completion)

Reply:
There's a few things that require it.

```
// Enable/disable semantic checking of JavaScript files. 
// Preferred quote style to use for quick fixes: `single` quotes, `double` quotes, or `auto` infer quote type from existing imports.
// Enable/disable suggestion diagnostics for JavaScript files in the editor. 
// Enable/disable automatic updating of import paths when you rename or move a file in VS Code.
// Enable/disable auto import suggestions.
```

I am a fan of using jsdoc comments to do a little typing so that things go smoother. Auto update paths and the auto import suggestions make things go quite faster. Just typing `css` to pull in emotion, or just writing a component I want and VSCode pulling it in is quite nifty.

JSX is just the newest one.